### ISSUE TITLE
Feature request: Make categories configurable

### ISSUE THREAD

**Issue by shynst** [spliit-app/spliit#101]
I didn't like the predefined categories back in the days in Splitwise and I still do not like them. So I went on and finally changed them in my instance in the database.

I also added a column pointing to the name of the icon to be used for that category (see [Lucide icon set](https://lucide.dev)).
The next obvious step would be to implement a UI to make categories really configurable.

### More ideas

- define a background color for each category group
- probably we would want to keep the two standard categories "General" and "Payment"
- when a category is about to be removed, warn if there are expenses referring to it. We could move them to "General" or let the user choose

### Going further...

How about being able to use a different set of categories for each group?

What's your opinion about it?

**Comment by Thain** [comment]
I agree a UI would be really helpful, but in lieu of that for now, can you describe how you changed the categories on your instance?

**Comment by shynst** [comment]
Here is my Category database:

<img width="511" alt="Screenshot 2024-07-29 at 22 01 59" src="https://github.com/user-attachments/assets/ec346ed4-c79f-4b76-8f92-2ba375010cde">

I could add a PR. But due to inactivity I implemented what I needed and thus my fork somehow diverged from this repo. You are of course free to look at my code: 

https://github.com/shynst/spliit

**Comment by TheToby1** [comment]
This would be great. I had written a list of categories I felt were missing, but found this in the issues and think it would be a much cleaner fix for my problem.
